bucket_map/src/bucket_api.rs

```diff
@@ -0,0 +1,134 @@
+use crate::bucket::Bucket;
+use crate::bucket_item::BucketItem;
+use crate::bucket_map::BucketMapError;
+use crate::bucket_stats::BucketMapStats;
+use crate::{MaxSearch, RefCount};
+use solana_sdk::pubkey::Pubkey;
+use std::ops::RangeBounds;
+use std::path::PathBuf;
+
+use std::sync::Arc;
+use std::sync::{RwLock, RwLockWriteGuard};
+
+type LockedBucket<T> = Arc<RwLock<Option<Bucket<T>>>>;
```

Comment:
As a TODO, we talked offline about removing the `Arc` from `LockedBucket` here. (For the next PR.)